preprocessors/caption.schema.json

```diff
@@ -0,0 +1,14 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema",
+    "$id": "https://image.a11y.mcgill.ca/preprocessors/caption.schema.json",
+    "type": "object",
+    "title": "Caption generator",
+    "description": "Caption generated by ca.mcgill.a11y.image.preprocessor.captionGenerator",
```

Comment:
Some quick notes:
- the reverse domain name ID does _not_ refer to the individual preprocessors, it refers to the data type described in this schema file that is used as an interface by one or more preprocessors that accomplish the same sort of task.
- Would recommend renaming to `captionGeneration` or something similar, as this is not generating captions itself
- If this type is meant to be limited to certain types of inputs, please specify. At the very least, clarify that it is a caption of a graphic.

Reply:
@JRegimbal I am not sure of your comment about the reverse domain? Is there any specific change that you are requesting wrt to the name?

Reply:
Caption is not generated by a schema. And the schema name should not include the word "generator" since schemas don't generate data, they describe them.